src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -996,19 +996,37 @@ public void connectFailed(URI arg0, SocketAddress arg1, IOException arg2) {}
+                TextChannel textChannel = DiscordSRV.getPlugin().getConsoleChannel();
+                return textChannel != null && textChannel.getGuild().getSelfMember().hasPermission(textChannel, Permission.MESSAGE_READ, Permission.MESSAGE_WRITE) ? textChannel : null;
+            }, config -> {
+                config.setLoggerNamePadding(config().getInt("DiscordConsoleChannelPadding"));
+                config.setLogLevels(EnumSet.copyOf(config().getStringList("DiscordConsoleChannelLevels").stream().map(String::toUpperCase).map(LogLevel::valueOf).collect(Collectors.toSet())));
+                config.mapLoggerName("net.minecraft.server.MinecraftServer", "Server");
+                config.mapLoggerNameFriendly("net.minecraft.server", s -> "Server/" + s);
+                config.mapLoggerNameFriendly("net.minecraft", s -> "Minecraft/" + s);
+                config.mapLoggerName("github.scarsz.discordsrv.dependencies.jda", s -> "DiscordSRV/JDA/" + s);
+                config.addTransformer(logItem -> true, s -> MessageUtil.strip(DiscordUtil.aggressiveStrip(s))); // strip formatting
+                config.addTransformer(logItem -> true, line -> {
+                    for (Map.Entry<Pattern, String> entry : consoleRegexes.entrySet()) {
+                        line = entry.getKey().matcher(line).replaceAll(entry.getValue());
+                        if (StringUtils.isBlank(line)) return null;
+                    }
+                    return line;
+                });
 
-            // start console message queue worker thread
-            if (consoleMessageQueueWorker != null) {
-                if (consoleMessageQueueWorker.getState() != Thread.State.NEW) {
-                    consoleMessageQueueWorker.interrupt();
-                    consoleMessageQueueWorker = new ConsoleMessageQueueWorker();
-                }
-            } else {
-                consoleMessageQueueWorker = new ConsoleMessageQueueWorker();
-            }
-            consoleMessageQueueWorker.start();
+                BiFunction<String, LogItem, String> placeholders = (key, item) -> {
+                    String name = config.padLoggerName(config.resolveLoggerName(item.getLogger()));
+                    String timestamp = TimeUtil.consoleTimeStamp(item.getTimestamp());
+                    return PlaceholderUtil.replacePlaceholdersToDiscord(config().getString(key))
+                            .replace("{date}", timestamp)
+                            .replace("{datetime}", timestamp)
+                            .replace("{name}", StringUtils.isNotBlank(name) ? " " + name : "")
+                            .replace("{level}", config.padLevelName(item.getLevel().name()));
+                };
+                config.setPrefixer(item -> placeholders.apply("DiscordConsoleChannelPrefix", item));
+                config.setSuffixer(item -> placeholders.apply("DiscordConsoleChannelSuffix", item));
+            }).attachLog4jLogging().schedule();
```

Comment:
Scheduling updates every 1 second will lead to a infinite rate limit loop if Discord ever has even the tiniest outage (already observed with Melanie)

Reply:
What would you suggest? 1.5 seconds?

Reply:
I'm ok with that, atleast that will clear up the rate limit queue after a while. 

Ideally avoiding building the "queue" up to begin with would be nice, but probably isn't very doable in practice